includes/admin/class-senseilms-plugin-updater.php

```diff
@@ -0,0 +1,231 @@
+		$res->version       = $remote->version;
+		$res->requires      = $remote->requires;
+		$res->tested        = $remote->tested;
+		$res->requires_php  = $remote->requires_php;
+		$res->last_updated  = $remote->last_updated;
+		$res->sections      = [
+			'description'  => $remote->sections->description,
+			'installation' => $remote->sections->installation,
+			'changelog'    => $remote->sections->changelog,
+		];
+		$res->download_link = $remote->download_url;
+		$res->banners       = [
+			'low'  => $remote->banners->{'1x'},
+			'high' => $remote->banners->{'2x'},
+		];
+
+		Sensei()->assets->enqueue( 'sensei-updater-styles', 'css/senseilms-licensing.css' );
+
+		return $res;
+	}
+
+	/**
+	 * Potentially injects the details for a new plugin version by checking against the remote server.
+	 * This is done by hooking into the `update_plugins` transient by using the `site_transient_update_plugins` hook.
+	 *
+	 * @param mixed $transient The plugin_update transient.
+	 *
+	 * @hooked site_transient_update_plugins See reference for `site_transient_transient`.
+	 *
+	 * @return mixed
+	 */
+	public function maybe_inject_custom_update_to_update_plugins_transient( $transient ) {
+
+		// Skip empty transients or if it was already set by Sensei Pro.
+		if ( empty( $transient ) || isset( $response[ $this->plugin_full_name ] ) ) {
+			return $transient;
+		}
+
+		$remote = $this->request_info();
+		if ( is_wp_error( $remote ) ) {
+			// Request failed so do not inject anything into the transient.
+			return $transient;
+		}
+
+		if (
+			$remote
+			&& version_compare( $this->version, $remote->version, '<' )
+			&& version_compare( get_bloginfo( 'version' ), $remote->requires, '>=' )
+			&& version_compare( PHP_VERSION, $remote->requires_php, '>=' )
+		) {
+
+			$res                                 = new stdClass();
+			$res->slug                           = $remote->slug;
+			$res->plugin                         = $this->plugin_full_name;
+			$res->new_version                    = $remote->version;
+			$res->tested                         = $remote->tested;
+			$res->package                        = $remote->download_url;
+			$res->icons                          = (array) $remote->icons;
+			$transient->response[ $res->plugin ] = $res;
+		}
+		return $transient;
+	}
+
+	/**
+	 * Helper function that retrieves the latest version information from the remote server if there is a valid license in the system.
+	 * This function caches remote response by using transients.
+	 *
+	 * @return array|WP_Error
+	 */
+	private function request_info() {
+		$cache_key = self::CACHE_KEY_PREFIX . $this->plugin_slug;
+		$remote    = get_transient( $cache_key );
+
+		if ( false === $remote ) {
+			$remote = wp_remote_get(
+				add_query_arg(
+					[
+						'plugin_slug' => $this->plugin_slug,
+						'ts'          => time(), // Adding some timestamp to workaround cache issues.
+					],
+					'https://senseilms.com/wp-json/plugin-updater/v1/info'
```

Comment:
What do you think about using here the `senseilms_licensing_api` filter?
I had a little bit of a hard time figuring out why I was not pointing to the correct environment :D

Reply:
You could also call directly `License_Manager::get_license_url` I know we are doing it somewhere else and I plan to move that into a different class and method name. But your change is also fine :)